catalyst_voices/lib/widgets/snackbar/voices_snackbar_type.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:catalyst_voices_brands/catalyst_voices_brands.dart';
+import 'package:catalyst_voices_localization/catalyst_voices_localization.dart';
+import 'package:flutter/material.dart';
+
+enum VoicesSnackBarType { info, success, warning, error }
+
+class _SnackBarData {
+  final IconData icon;
+  final Color? Function(VoicesColorScheme) iconColor;
+  final Color? Function(VoicesColorScheme) titleColor;
+  final Color? Function(VoicesColorScheme) backgroundColor;
+  final String Function(VoicesLocalizations) message;
+  final String Function(VoicesLocalizations) title;
+
+  const _SnackBarData({
+    required this.icon,
+    required this.iconColor,
+    required this.titleColor,
+    required this.backgroundColor,
+    required this.message,
+    required this.title,
+  });
+}
+
+extension VoicesSnackBarTypeExtension on VoicesSnackBarType {
```

Comment:
You could just define it as enum functions like so.

```dart
enum VoicesSnackBarType {
  info,
  success,
  warning,
  error;

  Color? backgroundColor(BuildContext context) {
    return switch (this) {
      VoicesSnackBarType.info => Theme.of(context).colors.primaryContainer,
      VoicesSnackBarType.success => throw UnimplementedError(),
      VoicesSnackBarType.warning => throw UnimplementedError(),
      VoicesSnackBarType.error => throw UnimplementedError(),
    };
  }
  ...
}

```

Then you may delete `_SnackBarData`

Reply:
@damian-molinski 

Using a map provides several  advantages:

1. **Performance Efficiency:**
   - **Map Lookup:** Maps provide O(1) average time complexity for lookups, making them faster for accessing properties.
   - **Switch Statements:** Each switch case has to be evaluated until a match is found, resulting in O(n) time complexity in the worst case.

2. **Code Maintainability:**
   - **Reduced Redundancy:** Properties are defined once in the map, eliminating the need for repetitive switch cases in multiple methods.
   - **Centralized Configuration:** All properties for each `VoicesSnackBarType` are stored in one place, making the code easier to read and update.
   - **Ease of Updates:** Adding or modifying a snackbar type or its properties requires a single change in the map, rather than updating multiple switch statements.

3. **Consistency and Scalability:**
   - **Consistent Definitions:** Centralized property definitions help ensure consistent values across the codebase.
   - **Scalable:** Easily extendable to add new snackbar types or properties without increasing code complexity.

Overall, using a map improves both performance and maintainability, resulting in more efficient and cleaner code.

Reply:
Roger that, thanks for explanation